### PR TITLE
support clean encoder/decoder, discard decode buffer

### LINKED ISSUE #241: 一个包含多个序列化值的字节数组进行反序列化的问题
一个字节数组bytes，里面包含了若干个hessian序列化后的值，可能是基础类型，也可能是自定义的Java对象。
1. 要怎么通过dubbo-go-hessian2将这些值反序列化出来？
2. 在字节数组中，这些值之间有没有明确的分界点？（用来实现指定反序列化某个值）

### PR DESCRIPTION
**Pull request by wongoo**
**What this PR does**:
support loop encoding&decoding

**Which issue(s) this PR fixes**:

Fixes #241

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
support loop encoding&decoding
```

### CODE REVIEW

File: hessian_demo/loop_encode_decode_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+	// -------- decode 1
+	res, err := d.Decode()
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.True(t, reflect.DeepEqual(u, res))
+
+	// -------- decode 2
+	d.Clean()
```
Comment (zonghaishang) [inline hessian_demo/loop_encode_decode_test.go]: Why is' clean 'called once per decode.
If you have a complex object type, such as an object with multiple instance arrays,  after the first instance is decoded, `clean` is called, will the decoding of subsequent instances fail?

File: hessian_demo/loop_encode_decode_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ */
+
+package hessian_demo
+
+import (
+	"reflect"
+	"testing"
+)
+
+import (
```
Comment (AlexStocks) [inline hessian_demo/loop_encode_decode_test.go:25]: move this import as the 3rd import block.

File: hessian_demo/loop_encode_decode_test.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	hessian "github.com/apache/dubbo-go-hessian2"
+)
+
+import (
```
Comment (AlexStocks) [inline hessian_demo/loop_encode_decode_test.go:29]: move this import as the 2rd import block.

File: hessian_demo/loop_encode_decode_test.go (hunk 1 above)
Comment (wongoo) [reply hessian_demo/loop_encode_decode_test.go]: a multiple instance arrays should be considered as a whole object, and will be all decoded in once.
A call fo `Clean` means a new start for a new object.

**Comment by wongoo** [comment]
@AlexStocks  there is a CI error `Error: 缺少参数 access_token`

**Comment by wongoo** [comment]
@AlexStocks @zonghaishang  this pr Expose the api `ReadByte`, add three new api `Encoder.Clean` , `Decoder.Clean` and `Decoder.Discard`. I think it can do more optimization for these api, but adding new api is big thing, i'd like to keep it simple, and we can do optimization afterwards.

**Review by AlexStocks — APPROVED** [review]

**Review by cityiron — APPROVED** [review]

**Review by zonghaishang — APPROVED** [review]

**Comment by cityiron** [comment]
> @AlexStocks there is a CI error `Error: 缺少参数 access_token`

PR  step can not get access_token

**Pull request merged**